prompt2model_demo.py

```diff
@@ -78,6 +78,45 @@ def print_logo():
     line_print(centered_ascii_art)
 
 
+def parse_model_size_limit(line: str, default_size=3e9) -> float:
+    """Parse the user input for the maximum size of the model.
+
+    Args:
+        line: The user input.
+        default_size: The default size to use if the user does not specify a size.
+    """
+    if len(line.strip()) == 0:
+        return default_size
+    model_units = {"B": 1e0, "KB": 1e3, "MB": 1e6, "GB": 1e9, "TB": 1e12, "PB": 1e15}
+    unit_disambiguations = {
+        "B": ["b", "bytes"],
+        "KB": ["Kb", "kb", "kilobytes"],
+        "MB": ["Mb", "mb", "megabytes"],
+        "GB": ["Gb", "gb", "gigabytes"],
+        "TB": ["Tb", "tb", "terabytes"],
+        "PB": ["Pb", "pb", "petabytes"],
+    }
+    unit_matched = False
+    for unit, disambiguations in unit_disambiguations.items():
+        all_disambiguations = [unit] + disambiguations
```

Comment:
Cleaner implementation:
for unit, disambiguations in unit_disambiguations.items():
            if unit == key or unit in disambiguations:
                unit = key
                break

Reply:
Done, thanks for the suggestion!